Latest systemtap versions require the probes.o file after all, per bug #557266.
This patch will be incorporated upstream soon.


diff -Naur postgresql-8.4.2.orig/src/backend/Makefile postgresql-8.4.2/src/backend/Makefile
--- postgresql-8.4.2.orig/src/backend/Makefile	2009-09-05 17:14:13.000000000 -0400
+++ postgresql-8.4.2/src/backend/Makefile	2010-01-20 17:21:37.000000000 -0500
@@ -20,13 +20,13 @@
 
 include $(srcdir)/common.mk
 
-# As of 9/2009:
-# * The probes.o file is necessary for dtrace support on Solaris.
-# * OS X's dtrace doesn't use it and doesn't even recognize the -G option.
-# * Systemtap's dtrace will take -G, but it produces a useless empty file.
-# So, build probes.o only on Solaris.
-# This will likely need adjustment as other platforms add dtrace support.
-ifeq ($(PORTNAME), solaris)
+# As of 1/2010:
+# The probes.o file is necessary for dtrace support on Solaris, and on recent
+# versions of systemtap.  (Older systemtap releases just produce an empty
+# file, but that's okay.)  However, OS X's dtrace doesn't use it and doesn't
+# even recognize the -G option.  So, build probes.o except on Darwin.
+# This might need adjustment as other platforms add dtrace support.
+ifneq ($(PORTNAME), darwin)
 ifeq ($(enable_dtrace), yes)
 LOCALOBJS += utils/probes.o
 endif
